Encoder/DefaultEncoder.php

```diff
@@ -29,10 +29,10 @@ public function __construct(JWSProviderInterface $jwsProvider)
     /**
      * {@inheritdoc}
      */
-    public function encode(array $payload)
+    public function encode(array $header, array $payload)
```

Comment:
Given this class can be an extension point, I think we should keep BC. We can do it by not changing the signature right now, handling the new argument only if passed and triggering a deprecation if it is not, saying that this method tzkes a new argument that will be mandatory in 3.0.
It can be done using func_num_args()/func_get_arg(), see https://github.com/symfony/symfony/blob/3.3/src/Symfony/Component/Console/Helper/QuestionHelper.php#L104 for inspiration